src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -77,10 +91,45 @@ public List<RawEntityTypeSummary> getEntityTypeSummary(Set<UUID> entityTypeIds)
       );
   }
 
+  private List<EntityTypeColumn> fetchColumnNamesForCustomFields(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getEntityTypeDefinition(entityTypeId)
+      .map(EntityType::getSourceView)
+      .orElseThrow(() -> new EntityTypeNotFoundException(entityTypeId));
+    return jooqContext
+      .select(field(REQUIRED_FIELD_NAME), field(REF_ID))
+      .from(sourceViewName)
+      .where(field(WHERE_CLAUSE).eq(CUSTOM_FIELD_TYPE))
+      .fetch()
+      .stream()
+      .map(row -> {
+        Object value = row.get(0);
+        Object extractedRefId = row.get(1);
+        assert value != null;
```

Comment:
I'm not sure how I feel about using `assert`... I've never actually seen it in production Java code before (mostly because it's so new), and it's new enough that I'm not sure if there's any real consensus in the Java community around it.